libc-test/build.rs

```diff
@@ -1093,3 +1027,121 @@ fn test_apple(target: &str) {
 
     cfg.generate("../src/lib.rs", "main.rs");
 }
+
+fn test_windows(target: &str) {
+    assert!(target.contains("windows"));
+    let gnu = target.contains("gnu");
+
+    let mut cfg = ctest::TestGenerator::new();
+    cfg.verbose_skip(true);
+
+    cfg.define("_WIN32_WINNT", Some("0x8000"));
+
+    // must be before windows.h:
+    headers! { cfg: "winsock2.h" }
+
+    headers! { cfg:
+        "direct.h",
+        "errno.h",
+        "fcntl.h",
+        "io.h",
+        "limits.h",
+        "locale.h",
+        "process.h",
+        "signal.h",
+        "stddef.h",
+        "stdint.h",
+        "stdio.h",
+        "stdlib.h",
+        "sys/stat.h",
+        "sys/types.h",
+        "sys/utime.h",
+        "time.h",
+        "wchar.h",
+        "windows.h",
+        "ws2ipdef.h",
```

Comment:
Ideally we would get rid of `winsock2.h` `windows.h` and `ws2ipdef.h` since they are part of Windows API and not the CRT. It would help prevent mistakes like https://github.com/rust-lang/libc/pull/1182